### PR TITLE
[BUG] mpdist window length

### PR DESCRIPTION
**Pull request by TonyBagnall**
fixes https://github.com/aeon-toolkit/aeon/issues/1498

the window length (called m for some reason) has as default value of 0, which means its set to a quarter of the length of the series. However, this check was happening after forming the windows, so it was trying to form windows with 0 length. 

I think it might be better to just use stumpy for this, m = None should be the default and k/threshold should definitely be a parameter. I will see if it makes a difference in a separate PR

### CODE REVIEW

**Comment by MatthewMiddlehurst** [comment]
could you make a change to run the CI again? If there are other problems would be good to create an issue.

File: aeon/distances/_mpdist.py [hunk 1]
```diff
@@ -69,6 +85,11 @@ def _mpdist(x: np.ndarray, y: np.ndarray, m: int) -> float:
     threshold = 0.05
     len_x = len(x)
     len_y = len(y)
+    if m == 0:
+        if len_x > len_y:
+            m = int(len_x / 4)
+        else:
+            m = int(len_y / 4)
```
Comment (baraline) [inline aeon/distances/_mpdist.py:92]: If you do this operation here but `len_x > 4 * len_y` you are going to throw errors down the line.

File: aeon/distances/_mpdist.py [hunk 2]
```diff
@@ -56,9 +63,18 @@ def mpdist(x: np.ndarray, y: np.ndarray, m: int = 0) -> float:
...
+            "to 1/4 of the length of the shortest time series"
+        )
     x = np.squeeze(x)
     y = np.squeeze(y)
-
+    if m > len(x) or m > len(y):
+        raise ValueError(
+            "subseries length must be less than or equal to the length "
+            "of both time series"
+        )
```
Comment (baraline) [inline aeon/distances/_mpdist.py:77]: This should be after the handling of the `m=0` case

**Review by baraline — CHANGES_REQUESTED** [review]

**Comment by baraline** [comment]
Overall I agree that having m=0 just feels weird as default value, None would be better !

**Review by baraline — APPROVED** [review]
LGTM to me now !

**Pull request merged**